lib/hash.cc

```diff
@@ -100,42 +98,25 @@ hash4(const void *data, unsigned int size)
     return h % size;
 }
 
-/**
- *  hash_create - creates a new hash table, uses the cmp_func
- *  to compare keys.  Returns the identification for the hash table;
- *  otherwise returns a negative number on error.
- */
-hash_table *
-hash_create(HASHCMP * cmp_func, int hash_sz, HASHHASH * hash_func)
-{
-    hash_table *hid = (hash_table *)xcalloc(1, sizeof(hash_table));
-    if (!hash_sz)
-        hid->size = (unsigned int) DEFAULT_HASH_SIZE;
-    else
-        hid->size = (unsigned int) hash_sz;
-    /* allocate and null the buckets */
-    hid->buckets = (hash_link **)xcalloc(hid->size, sizeof(hash_link *));
-    hid->cmp = cmp_func;
-    hid->hash = hash_func;
-    hid->next = NULL;
-    hid->current_slot = 0;
-    return hid;
+hash_table::hash_table(HASHCMP *cmp_func, HASHHASH *hash_func, int hash_sz)
+    : count(0), next(nullptr), size(hash_sz),
+      current_slot(0), hash(hash_func), cmp(cmp_func) {
+    buckets = static_cast<hash_link **>(xcalloc(size, sizeof(hash_link *)));
 }
 
 /**
- *  hash_join - joins a hash_link under its key lnk->key
- *  into the hash table 'hid'.
+ *  join a hash_link under its key lnk->key
+ *  into the hash table.
  *
  *  It does not copy any data into the hash table, only links pointers.
  */
 void
-hash_join(hash_table * hid, hash_link * lnk)
-{
+hash_table::hash_join(hash_link *lnk) {
```

Comment:
Please keep the Squid coding style with brackets on a separate line.
```suggestion
hash_table::hash_join(hash_link *lnk)
{
```